src/components/validation/resources/http.rs

```diff
@@ -312,14 +353,25 @@ fn spawn_output_http_server(
 
     tokio::spawn(async move {
         resource_started.mark_as_done();
-        debug!("HTTP server external output resource started.");
+        info!("HTTP server external output resource started.");
 
+        // Wait for the runner to tell us to shutdown
         resource_shutdown_rx.wait().await;
-        _ = http_server_shutdown_tx.send(());
+
+        // ensure we've processed all events we expect to receive from the sink, if any.
+        if expected_output_events > 0 {
+            server_sent_all_rx.recv().await;
+        }
```

Comment:
Separate from my comment above about restructuring this somewhat to make the logic more durable... one thing that is _perhaps_ worth keeping in mind (but is just a comment here, not a thing to fix/address) is that this _does_ break the ability for the validation runner to gracefully kill tests, since we'll be stuck waiting here until N events are processed before we ever instruct the HTTP server to actually stop.

Reply:
Mmm I see what you mean yeah. We'd not get the output from the runner to know what metrics we did capture and where the discrepancies were. I think this could be solved by a timeout mechanism, I'll tack on on to the next PR since this one is approved.

Reply:
resolved in latest changes, we now have a timeout